cmd/kots/cli/check-version.go

```diff
@@ -17,7 +18,10 @@ import (
 
 // Checks the KOTS CLI version against the API version
 func cliVersionCheck() error {
-	cliVersion := buildversion.Version()
+	cliSemver, err := semver.ParseTolerant(buildversion.Version())
```

Comment:
Just refactored the version check method to make it more testable and added a bunch of cases.  This will hopefully give an idea of what will produce a warning.